freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java

```diff
@@ -151,4 +155,20 @@ private String readCatalogFromFile(String catalogUrl) throws IOException {
         File customCatalogFile = new File(etcConfigDir, catalogUrl);
         return FileReaderUtils.readFileFromPath(customCatalogFile.toPath());
     }
+
+    private Versions getVersions(ImageCatalog catalog) {
+        if (catalog.getVersions() == null || catalog.getVersions().getFreeIpaVersions() == null) {
+            LOGGER.debug("FreeIPA versions are missing from the image catalog.");
```

Comment:
For clarity I would extend the message to something like "FreeIPA versions are missing from the image catalog, generating it from advertised flags."